sandbox/brooklyn-sandbox-test-framework/src/main/java/org/apache/brooklyn/test/framework/TestHttpCallImpl.java

```diff
@@ -0,0 +1,98 @@
+package org.apache.brooklyn.test.framework;
+
+import com.google.api.client.util.Objects;
+import com.google.common.collect.Lists;
+import org.apache.brooklyn.api.location.Location;
+import org.apache.brooklyn.core.entity.AbstractEntity;
+import org.apache.brooklyn.core.entity.lifecycle.Lifecycle;
+import org.apache.brooklyn.core.entity.lifecycle.ServiceStateLogic;
+import org.apache.brooklyn.util.core.flags.TypeCoercions;
+import org.apache.brooklyn.util.exceptions.Exceptions;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Collection;
+import java.util.Map;
+
+import static org.apache.brooklyn.util.http.HttpUtils.*;
+
+/**
+ * {@inheritDoc}
+ */
+public class TestHttpCallImpl extends AbstractEntity implements TestHttpCall {
+
+    private static final Logger LOG = LoggerFactory.getLogger(TestHttpCallImpl.class);
+
+    /**
+     * {@inheritDoc}
+     */
+    public void start(Collection<? extends Location> locations) {
+        ServiceStateLogic.setExpectedState(this, Lifecycle.STARTING);
+        final String url = getConfig(TARGET_URL);
+        final Map assertions = getConfig(ASSERTIONS);
+        try {
+            checkAssertions(url, assertions);
+            sensors().set(SERVICE_UP, true);
+            ServiceStateLogic.setExpectedState(this, Lifecycle.RUNNING);
+        } catch (Throwable t) {
+            LOG.info("Url [{}] test failed", url);
+            sensors().set(SERVICE_UP, false);
+            ServiceStateLogic.setExpectedState(this, Lifecycle.ON_FIRE);
+            throw Exceptions.propagate(t);
+        }
+    }
+
+    /**
+     * Tests HTTP Request reponse matches assertions
+     * <p>
+     * Supported keys in the <code>assertions</code> {@link Map} include:
+     * <ul>
+     * <li>string - simple string match
+     * <li>regex - uses {@link java.lang.String#matches(String)}, if the url returns a multi-line response you should
+     * use the embedded dotall flag expression <code>(?s)</code> in your regex.
+     * <li>status - HTTP status code
+     * </ul>
+     * Wraps the {@link org.apache.brooklyn.util.http.HttpUtils} immediate assertion methods.
+     * <p>
+     * See the test/resources directory for examples.
+     *
+     * @param url        The target URL to be tested
+     * @param assertions The map of assertions
+     */
+    private void checkAssertions(final String url, final Map<?, ?> assertions) {
+
+        for (final Map.Entry<?, ?> entry : assertions.entrySet()) {
+            if (Objects.equal(entry.getKey(), "regex")) {
+                LOG.info("Testing if url [{}] matches regex [{}]",
+                        new Object[]{url, entry.getValue()});
+                assertContentMatches(url, TypeCoercions.coerce(entry.getValue(), String.class));
+            } else if (Objects.equal(entry.getKey(), "string")) {
```

Comment:
agree with @m4rkmckenna that `contains` would be more meaningful than `string` in this context